src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -58,29 +62,70 @@ export class FilterTimestamp extends HTMLElement {
     );
   }
 
-  onClearFilter = () => {
-    this.startTimeLabel.textContent = 'From Beginning';
-    this.stopTimeLabel.textContent = 'Up to End';
-    this.filterContainer.hidden = true;
+  onClearFilter(clearStart, clearStop) {
+    if (clearStart) {
+      this.startTimeLabel.textContent = 'From Beginning';
+      this.clearStartTimeBtn.hidden = true;
+      delete this.timeStamp.start;
+    }
+
+    if (clearStop) {
+      this.stopTimeLabel.textContent = 'Up to End';
+      this.clearStopTimeBtn.hidden = true;
+      delete this.timeStamp.stop;
+    }
+
+    if (!clearStart && !clearStop) {
+      this.startTimeLabel.textContent = 'From Beginning';
+      this.stopTimeLabel.textContent = 'Up to End';
+      this.timeStamp = {};
+    }
+
+    if (Object.keys(this.timeStamp).length === 0) {
+      this.timeStamp = null;
+    }
```

Comment:
@rpl I had to use `Object.keys(this.timeStamp).length === 0` here.

Reply:
@atiqueahmedziad That's fine, I don't like it but it is not such a big deal.

But please be consistent with the time passed to the parameters, if they are supposed to be boolean, pass a boolean for them or omit them completely and set a default value for the variable.

e.g instead of
```
      switch (event.target) {
        case this.clearFilterBtn:
          this.onClearFilter(null, null);
          break;
        case this.clearStartTimeBtn:
          this.onClearFilter(true, null);
          break;
        case this.clearStopTimeBtn:
          this.onClearFilter(null, true);
          break;
      }
```

we could define this function as:

```
  onClearFilter(clearStart = true, clearStop = true) {
    ...
  }
```

and then call it without any parameter if we want to clear both start and stop, and with two explicit boolean parameters if we want to clear only one of them:

```
      switch (event.target) {
        case this.clearFilterBtn:
          this.onClearFilter();
          break;
        case this.clearStartTimeBtn:
          this.onClearFilter(true, false);
          break;
        case this.clearStopTimeBtn:
          this.onClearFilter(false, true);
          break;
      }
```